### PR TITLE
(fix): update modeling example

### PR DESCRIPTION
**Pull request by sjmignot**

### CODE REVIEW

**Review by AndyMoreland — APPROVED** [review]

File: 12_model/1_model.py [hunk 1]
```diff
@@ -1,31 +1,43 @@
+"""Running Predictive Models in Chalk
+
+In this exmaple, we are using a logistic regression model to predict
+whether a user will churn from our video streaming service. This example
+assumes, we have already trained a simple model on age, num_friends and
```
Comment (emarx) [inline 12_model/1_model.py:5]:
```suggestion
assumes we have already trained a simple model on age, num_friends and
```

File: 12_model/1_model.py [hunk 2]
```diff
@@ -1,31 +1,43 @@
...
+viewed_minutes features for a number of our users (a dummy model—
+named 'churn_model.skops'—can be found in this file's directory).
```
Comment (emarx) [inline 12_model/1_model.py:7]:
```suggestion
`viewed_minutes` features for a number of our users (a dummy model—
named `'churn_model.skops'`—can be found in this file's directory).
```

File: 12_model/1_model.py [hunk 3]
```diff
@@ -1,31 +1,43 @@
...
+import os
 
 
 @features
 class User:
     id: str
     age: int
     num_friends: int
-    favorite_categories: Sequence[str]
+    favorite_categories: List[str]
```
Comment (emarx) [inline 12_model/1_model.py:24]: i think this feature is unused? could we delete the ones that aren't relevant?

File: 12_model/1_model.py [hunk 4]
```diff
@@ -1,31 +1,43 @@
...
     viewed_minutes: float
     location: str
-    model_prediction: float
+    probability_of_churn: float
```
Comment (emarx) [inline 12_model/1_model.py:27]: let's add a comment here that describes that we're going to predict this one

File: 12_model/1_model.py [hunk 5]
```diff
@@ -39,14 +51,25 @@ def predict(self, data: np.array):
...
 
 
-# model has been trained and saved in local chalk directory
-model = PredictionModel("model.skops")
+# the model has been trained and saved in local chalk directory
+churn_model = PredictionModel(
+    os.path.join(
+        os.environ.get("TARGET_ROOT"),
+        "churn_model.skops"
+    )
```
Comment (emarx) [inline 12_model/1_model.py:59]: target root is only available in the deployed environment, so the path join should be in the `_model` function. just pass in the name of the file from the root. also let's add a comment around the rand() string TARGET_ROOT linking to the page that I sent over in slack

File: 12_model/1_model.py [hunk 6]
```diff
@@ -1,52 +1,71 @@
...
 @features
 class User:
     id: str
     age: int
     num_friends: int
-    favorite_categories: Sequence[str]
     viewed_minutes: float
-    location: str
-    model_prediction: float
+    probability_of_churn: float  # this is the feature we will be predicting
```
[the remark this answers is not in the document]
Comment (sjmignot) [reply 12_model/1_model.py:25]: Gotcha, i didn't know that functionality existed, very cool

File: 12_model/1_model.py [hunk 7]
```diff
@@ -1,52 +1,79 @@
...
     age: int
     num_friends: int
-    favorite_categories: Sequence[str]
     viewed_minutes: float
-    location: str
-    model_prediction: float
+    # this is a feature based on a model prediction: we will a
+    # model we previous trained on a users age, num_friends, and
+    # viewed_minutes to predict how likely they are to churn in
+    # a chalk resolver.
```
Comment (emarx) [inline 12_model/1_model.py:32]:
```suggestion
    # a Chalk resolver.
```

File: 12_model/1_model.py [hunk 8]
```diff
@@ -1,52 +1,79 @@
...
+    """
     id: str
     age: int
     num_friends: int
-    favorite_categories: Sequence[str]
     viewed_minutes: float
-    location: str
-    model_prediction: float
+    # this is a feature based on a model prediction: we will a
+    # model we previous trained on a users age, num_friends, and
```
Comment (emarx) [inline 12_model/1_model.py:30]: the text here is a little wonky. `we will a model we ...`. also maybe the lead in slightly awkward phrasing?

File: 12_model/1_model.py [hunk 9]
```diff
@@ -1,52 +1,84 @@
...
-from typing import Sequence
+from typing import List
+import os
 
 
 @features
 class User:
+    """This is a User for our streaming streaming platform. We keep track
+    of a user's age, how many friends they have on our platform, and how
+    many minutes of video they have viewed.
```
Comment (sjmignot) [inline 12_model/1_model.py:23]:
```suggestion
    many minutes of content they have viewed.
```

File: 12_model/1_model.py [hunk 10]
```diff
@@ -1,52 +1,84 @@
...
+    # We run this prediction in the 'get_user_churn_probability'
+    # resolver below.
+    probability_of_churn: float
 
 
 class PredictionModel:
     """
-    from sklearn.ensemble import RandomForestRegressor
+    # Previously, we trained a model on our user data. This
+    # model has been saved to our local chalk directory, next to
```
Comment (sjmignot) [inline 12_model/1_model.py:39]:
```suggestion
    # model has been saved to our local chalk directory next to
```

File: 12_model/1_model.py [hunk 11]
```diff
@@ -1,52 +1,84 @@
...
+
+        return load(filepath, trusted=True)
 
     def predict(self, data: np.array):
         return self._model.predict(data)
 
 
-# model has been trained and saved in local chalk directory
-model = PredictionModel("model.skops")
+# the model has been trained and saved in local chalk directory
```
Comment (sjmignot) [inline 12_model/1_model.py:70]:
```suggestion
# the model has been trained and saved in our local Chalk directory
```

File: 12_model/1_model.py [hunk 12]
```diff
@@ -1,52 +1,84 @@
...
+def get_user_churn_probability(
+    age: User.age,
+    num_friends: User.num_friends,
+    viewed_minutes: User.viewed_minutes,
+) -> User.probability_of_churn:
+    """
+    This resolver runs a model that has been trained on a user's age, num_friends
+    and viewed_minutes. It returns a platform churn prediction.
+    """
+    return churn_model.predict(np.array([[age, num_friends, viewed_minutes]]))
```
Comment (sjmignot) [inline 12_model/1_model.py:84]:
```suggestion
    return churn_model.predict(np.array([[age, num_friends, viewed_minutes]])).squeeze()
```

**Pull request merged**